### ISSUE TITLE
BUG: geocoder default no longer valid

### ISSUE THREAD

**Issue by nickeubank** [geopandas/geopandas#2059]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of geopandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of geopandas.

---

Related to #2005, which fixed for tests but not the library default. 

#### Code Sample, a copy-pastable example

```python
import geopandas as gpd
geocoded = gpd.tools.geocode(df.address.sample(20))
```
#### Problem description

```python 
GeocoderNotFound: Unknown geocoder 'geocodefarm'; options are: dict_keys(['algolia', 'arcgis', 'azure', 'baidu', 'baiduv3', 'banfrance', 'bing', 'databc', 'geocodeearth', 'geocodio', 'geonames', 'google', 'googlev3', 'geolake', 'here', 'herev7', 'ignfrance', 'mapbox', 'mapquest', 'maptiler', 'nominatim', 'opencage', 'openmapquest', 'pickpoint', 'pelias', 'photon', 'liveaddress', 'tomtom', 'what3words', 'what3wordsv3', 'yandex'])
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
~/opt/miniconda3/lib/python3.9/site-packages/geopy/geocoders/__init__.py in get_geocoder_for_service(service)
    317     try:
--> 318         return SERVICE_TO_GEOCODER[service.lower()]
    319     except KeyError:

KeyError: 'geocodefarm'

During handling of the above exception, another exception occurred:

GeocoderNotFound                          Traceback (most recent call last)
/var/folders/tj/s8f2_ks15h315z5thvtnhz8r0000gp/T/ipykernel_66753/3206994461.py in <module>
      1 import geopandas as gpd
----> 2 geocoded = gpd.tools.geocode(df.address.sample(20))

~/opt/miniconda3/lib/python3.9/site-packages/geopandas/tools/geocoding.py in geocode(strings, provider, **kwargs)
     69         throttle_time = _get_throttle_time(provider)
     70 
---> 71     return _query(strings, True, provider, throttle_time, **kwargs)
     72 
     73 

~/opt/miniconda3/lib/python3.9/site-packages/geopandas/tools/geocoding.py in _query(data, forward, provider, throttle_time, **kwargs)
    136 
    137     if isinstance(provider, str):
--> 138         provider = get_geocoder_for_service(provider)
    139 
    140     coder = provider(**kwargs)

~/opt/miniconda3/lib/python3.9/site-packages/geopy/geocoders/__init__.py in get_geocoder_for_service(service)
    318         return SERVICE_TO_GEOCODER[service.lower()]
    319     except KeyError:
--> 320         raise GeocoderNotFound(
    321             "Unknown geocoder '%s'; options are: %s" %
    322             (service, SERVICE_TO_GEOCODER.keys())

GeocoderNotFound: Unknown geocoder 'geocodefarm'; options are: dict_keys(['algolia', 'arcgis', 'azure', 'baidu', 'baiduv3', 'banfrance', 'bing', 'databc', 'geocodeearth', 'geocodio', 'geonames', 'google', 'googlev3', 'geolake', 'here', 'herev7', 'ignfrance', 'mapbox', 'mapquest', 'maptiler', 'nominatim', 'opencage', 'openmapquest', 'pickpoint', 'pelias', 'photon', 'liveaddress', 'tomtom', 'what3words', 'what3wordsv3', 'yandex'])
```

#### Output of ``geopandas.show_versions()``

<details>

>>> gpd.show_versions()

SYSTEM INFO
-----------
python     : 3.9.6 | packaged by conda-forge | (default, Jul  6 2021, 08:54:05)  [Clang 11.1.0 ]
executable : /Users/Nick/opt/miniconda3/bin/python
machine    : macOS-11.5.1-x86_64-i386-64bit

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : 3.9.1
GEOS lib   : /Users/Nick/opt/miniconda3/lib/libgeos_c.dylib
GDAL       : 3.3.1
GDAL data dir: /Users/Nick/opt/miniconda3/share/gdal
PROJ       : 8.0.1
PROJ data dir: /Users/Nick/opt/miniconda3/share/proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 0.9.0
pandas     : 1.3.0
fiona      : 1.8.20
numpy      : 1.21.1
shapely    : 1.7.1
rtree      : 0.9.7
pyproj     : 3.1.0
matplotlib : 3.4.2
mapclassify: 2.4.3
geopy      : 2.2.0
psycopg2   : None
geoalchemy2: None
pyarrow    : None
pygeos     : None

</details>

**Comment by martinfleis** [comment]
Hi, thanks for the report. This has been fixed in #2007. We now use Photon as a default.

**Comment by nickeubank** [comment]
Thanks!
